src/lints/struct_now_doc_hidden.ron

```diff
@@ -0,0 +1,50 @@
+SemverQuery(
+    id: "struct_now_doc_hidden",
+    human_readable_name: "pub struct is now #[doc(hidden)]",
+    description: "A pub struct is now hidden from documentation effectively removing it from the crate's public api.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/rustdoc/write-documentation/the-doc-attribute.html#hidden"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Struct {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        struct_type @output @tag
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Struct {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        struct_name: name @output
+                        struct_type @filter(op: "=", value: ["%struct_type"])
```

Comment:
I wonder if we should care that the struct remains the same kind of struct (plain, tuple, or unit struct) across this change.

Normally we use this to make sure we don't trigger multiple lints, like "struct pub field removed" _and_ "struct changed kind." But whether the struct is public API or not seems orthogonal to whether it changed kind or has fields, so I'm having a hard time imagining a duplicate lint situation here.

What do you think?

Reply:
Yeah, you're right.